### ISSUE TITLE
[Feature Request] Fooyin doesn't recognize/handle MKA (Matroska) files.

### ISSUE THREAD

**Issue by Continous** [ludouzi/fooyin#125]
Fooyin should technically support Matroska files out of the box from FFMPEG, however Fooyin doesn't automatically consider .mka files as audio files.

Personally, I use it as it allows essentially arbitrary audio encodings, and allows me to avoid re-encoding when extracting audio from a video file (as would be necessary when obtaining audio from a Blu-ray audio disc). There could be other useful applications for an MKA, such as perhaps containing lyrics information within the file.

**Comment by ludouzi** [comment]
You're right in that there won't be any issues in terms of playback. The only issue as you found out is the metadata/file properties. 

Matroska files are not currently supported because TagLib doesn't support reading them. This will change in time as I'm planning to support additional tag readers as plugins. Seeing as we're already using it, an FFmpeg-based reader would be the first to be implemented, as it can read the file metadata of a lot of formats (including Matroska). 

I'll look into this soonish.

**Comment by kode54** [comment]
You may find FFmpeg a bit troublesome, as at least ID3v2 requires a full ID3v2 parser for the tag chunk. And like album artwork, it requires opening the file with avformat and walking frames of the file on the non-audio streams it enumerates at open time. Not that you need to use ID3v2 since you're already using TagLib for MP3 files, but just thought I'd let you know about interaction possibly with other exotic formats with embedded tags.

Artwork will identify itself as MJPEG streams or similar, and contain a single packet "frame" of the artwork data.

If/when you get around to doing streams, it will require similar metadata polling to deal with stream switching tags for the Ogg based formats.

Chapters should also be fun for FFmpeg, but that's easy enough, since FFmpeg exposes those as timestamped ranges of the file. Gaplessness of some formats in FFmpeg also depends on parsing the chapter fields, as even single song files will encode data there as a single "chapter" that starts after the decoded beginning of the audio clip, indicating how much to lop off the start, and how long the clip should be.

**Comment by alekksander** [comment]
@Continous  it's mid 2026 and matroska seem to be handled properly. Please close or elaborate further.

**Comment by Continous** [comment]
> [@Continous](https://github.com/Continous) it's mid 2026 and matroska seem to be handled properly. Please close or elaborate further.

There are still some .MKA handling issues. Below are a list of features that I would expect from full support;

- [x] Playback.
- [x] Chapters handling.
- [x] Album Art Handling.
- [ ] Track selection.
- [ ] Track info handling. (Including handling "Commentary" flagging appropriately.)

These are basic features of the format and why one would use this format over, say, FLAC. As far as I can tell from the latest Flatpak release, there is just track selection left to manage. If there have been updates since then I'd like to know as the latest NixOS package is quite outdated.
